test/profiler/test_profiler.py

```diff
@@ -883,18 +895,19 @@ def join_threads(context: bool):
         ]
         tid_counts = collections.Counter([node.start_tid for node in nodes])
 
+        diff_tid_counts = tid_counts - base_tid_counts
```

Comment:
Consider this corner case: What happens if one of the base_tid_counts threads finishes before we get to this point? for example, base_tid_counts = 1, and this test creates 2 expected_threads. If base thread finishes, the tid_counts = 2, and diff_tid_counts = 1. That would be less than expected_threads = 2.

Reply:
That is true. I added this in because the threads that are causing this failure are persisting indefinitely (until the main thread closes of course). I am not sure how we would be able to reasonable cover this case